app script

karolokninski opened the pull request.

karolokninski committed:
app script

kepucino commented on app.py:
```diff
@@ -0,0 +1,159 @@
+from datetime import datetime
+
+# %%
+data_path = '/home/karol/electricity-prices-app/prices.csv'
```
możliwe, że trzeba zmienić ścieżkę
nie wiem na jaką

karolokninski committed:
Dash app

karolokninski committed:
app with daily data update

karolokninski committed:
app with daily data update

karolokninski committed:
app with daily data update

karolokninski committed:
app with daily data update

karolokninski committed:
modified:   app.py
	modified:   dash.ipynb
	modified:   prices.csv

karolokninski committed:
ci: add Azure Static Web Apps workflow file
on-behalf-of: @Azure opensource@microsoft.com

karolokninski committed:
modified:   Dockerfile
	modified:   app.py
	modified:   assets/styles.css
	modified:   dash.ipynb
	modified:   prices.csv
	modified:   requirements.txt

karolokninski committed:
Merge branch 'app' of https://github.com/kepucino/electricity-price-analysis into app

karolokninski committed:
db update script

karolokninski committed:
modified:   app.py
	modified:   dash.ipynb

karolokninski committed:
new file:   app2.py
	new file:   assets/favicon.ico
	new file:   s.py
	new file:   scraper.py

karolokninski committed:
solar production calculation

karolokninski committed:
Delete s.py

karolokninski committed:
request data from py script

karolokninski committed:
Merge branch 'app' of https://github.com/kepucino/electricity-price-analysis into app

karolokninski committed:
PV production API

karolokninski committed:
PV production API

konradsitarz commented on README.MD:
```diff
@@ -0,0 +1,57 @@
+    'Tue, 11 Jul 2023 14:00:00 GMT',
+    'Tue, 11 Jul 2023 15:00:00 GMT',
+    'Tue, 11 Jul 2023 16:00:00 GMT']
+}
```
Change dates to format  
```
# YYYY-MM-DD HH:MM:SS
pd.Timestamp('2017-03-22 15:16:45')
```

konradsitarz commented on README.MD:
add info about auth header

konradsitarz commented on api.py:
```diff
@@ -0,0 +1,71 @@
+@app.route('/estimate', methods=['POST'])
+@require_token
+def estimate():
+    request_data = request.json
```
1. parsing/data validation here 
2. Calling the weather api
3. Calculating results

```
# throws errors and sends validation errors back in json 
# can be in module validation
params = validation.validate_estimate_params(request_data) 

# In module services, params can be kwargs
weather_df = services.load_weather_data(params)

# Calculates energy production and profit from selling all energy
production = simulation.calculate_production(weather_df, params)
profit = simulation.calculate_profit(production)

# Serialize objects with serializer
# Ensures that the right data is returned
# Does something like below
# `json.dumps( {timestamp, production, profit})`
result = serializer.serialize_estimation_results(prodcution, profit)
return result
```

karolokninski committed:
ability to save emails, profit from estimate

karolokninski committed:
remove unused files
